WordPressCom-Stats-iOS/StatsPostDetailsTableViewController.m

```diff
@@ -0,0 +1,523 @@
+static NSString *const StatsTableGraphCellIdentifier = @"GraphRow";
+static NSString *const StatsTableNoResultsCellIdentifier = @"NoResultsRow";
+
+@interface StatsPostDetailsTableViewController () <WPStatsGraphViewControllerDelegate>
+
+@property (nonatomic, strong) NSMutableArray *sections;
+@property (nonatomic, strong) NSMutableDictionary *sectionData;
+@property (nonatomic, strong) WPStatsGraphViewController *graphViewController;
+@property (nonatomic, strong) NSDate *selectedDate;
+@property (nonatomic, assign) BOOL isRefreshing;
+@end
+
+@implementation StatsPostDetailsTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    [WPAnalytics track:WPAnalyticsStatStatsSinglePostAccessed];
+    
+    self.sections = [@[@(StatsSectionPostDetailsLoadingIndicator), @(StatsSectionPostDetailsGraph), @(StatsSectionPostDetailsMonthsYears), @(StatsSectionPostDetailsAveragePerDay), @(StatsSectionPostDetailsRecentWeeks)] mutableCopy];
+    self.sectionData = [NSMutableDictionary new];
+
+    self.tableView.tableHeaderView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 0, 20.0f)];
+    self.tableView.backgroundColor = [WPStyleGuide itsEverywhereGrey];
+    self.tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
+    [self.tableView registerClass:[StatsTableSectionHeaderView class] forHeaderFooterViewReuseIdentifier:StatsTableSectionHeaderSimpleBorder];
+    
+    [self setupRefreshControl];
+    
+    self.graphViewController = [WPStatsGraphViewController new];
+    self.graphViewController.allowDeselection = NO;
+    self.graphViewController.graphDelegate = self;
+    [self addChildViewController:self.graphViewController];
+    [self.graphViewController didMoveToParentViewController:self];
+    
+    self.title = self.postTitle;
+    
+    self.sectionData[@(StatsSectionPostDetailsGraph)] = [StatsVisits new];
+    self.sectionData[@(StatsSectionPostDetailsMonthsYears)] = [[StatsGroup alloc] initWithStatsSection:StatsSectionPostDetailsMonthsYears andStatsSubSection:StatsSubSectionNone];
+    self.sectionData[@(StatsSectionPostDetailsAveragePerDay)] = [[StatsGroup alloc] initWithStatsSection:StatsSectionPostDetailsAveragePerDay andStatsSubSection:StatsSubSectionNone];
+    self.sectionData[@(StatsSectionPostDetailsRecentWeeks)] = [[StatsGroup alloc] initWithStatsSection:StatsSectionPostDetailsRecentWeeks andStatsSubSection:StatsSubSectionNone];
+    [self.tableView reloadData];
+}
+
+- (void)viewDidAppear:(BOOL)animated
+{
+    [super viewDidAppear:animated];
+    
+    [self retrieveStats];
+}
+
+
+- (void)viewWillDisappear:(BOOL)animated
+{
+    [super viewWillDisappear:animated];
+    
+    [self abortRetrieveStats];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return 4 + (self.isRefreshing ? 1 : 0);
```

Comment:
Actually I have a collection of sections and I should just be returning the count of that array.